engine/schema/src/com/cloud/vm/dao/UserVmDaoImpl.java

```diff
@@ -192,6 +192,18 @@ void init() {
                 JoinBuilder.JoinType.INNER);
         AccountDataCenterVirtualSearch.done();
 
+        SearchBuilder<NicVO> nicSearchByNetwork = _nicDao.createSearchBuilder();
```

Comment:
@nitin-maharana while your fix may be valid, it seems to be breaking some test_nic and test_routers test cases. The issue seems to be another `nicSearch` on line 184 which perhaps is used in most cases. You may remove `nicSearchByNetwork` which is not useful or used at all depending on the original code when UserVmSearch is null or not, when calling the method `listByNetworkIdAndStates`.  Depending on the test results, I may revert your original fix, but kindly re-submit a newer change and I can help run tests for you. Thanks.